webgpu.c

```diff
@@ -735,9 +735,70 @@ WGPUCommandEncoder wgpuDeviceCreateCommandEncoder(WGPUDevice device, WGPUCommand
     return command_encoder_struct;
 }
 
-// WGPUComputePipeline wgpuDeviceCreateComputePipeline(WGPUDevice device, WGPUComputePipelineDescriptor const* descriptor)
-// {
-// }
+WGPUComputePipeline wgpuDeviceCreateComputePipeline(WGPUDevice device, WGPUComputePipelineDescriptor const* descriptor)
+{
+    if (!device || !descriptor) unreachable();
+
+    wasi_webgpu_webgpu_gpu_layout_mode_t layout_wasi = {};
+    if (!descriptor->layout) {
+        layout_wasi.tag = WASI_WEBGPU_WEBGPU_GPU_LAYOUT_MODE_AUTO;
+    } else {
+        layout_wasi.tag = WASI_WEBGPU_WEBGPU_GPU_LAYOUT_MODE_SPECIFIC;
+        layout_wasi.val.specific = wasi_webgpu_webgpu_borrow_gpu_pipeline_layout(descriptor->layout->pipeline_layout);
+    }
+
+    imports_option_string_t entry_point_wasi;
```

Comment:
I think we should be defensive here. Initialize so that later on line 785 it's at least initialized when !entry_point_wasi.is_some

imports_option_string_t  entry_point_wasi = {};